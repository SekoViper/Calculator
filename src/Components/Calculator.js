/* eslint-disable no-plusplus */
import React from 'react';
import calculate from '../Logic/calculate';

export default class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.buttonClick = this.buttonClick.bind(this);
  }

  createDigits = () => {
    const digits = [];
    for (let i = 1; i < 10; i++) {
      digits.push(
        <button type="button" key={i} onClick={this.buttonClick}>{i}</button>,
      );
    }
    return digits;
  };

  buttonClick(element) {
    this.setState((state) => calculate(state, element.target.textContent));
  }

  render() {
    const { next } = this.state;
    const { total } = this.state;
    return (
      <div className="calculator">
        <div className="display">
          <span className="results">{next || total || 0}</span>
        </div>

        <div className="buttons">
          <div className="digits">
            <button type="button" onClick={this.buttonClick}>AC</button>
            <button type="button" onClick={this.buttonClick}>+/-</button>
            <button type="button" onClick={this.buttonClick}>%</button>
            {this.createDigits()}
            <button className="span-row" type="button" onClick={this.buttonClick}>0</button>
            <button className="s" type="button" onClick={this.buttonClick}>.</button>
          </div>

          <div className="operators">
            <button className="operator" type="button" onClick={this.buttonClick}>/</button>
            <button className="operator" type="button" onClick={this.buttonClick}>+</button>
            <button className="operator" type="button" onClick={this.buttonClick}>-</button>
            <button className="operator" type="button" onClick={this.buttonClick}>x</button>
            <button className="operator" type="button" onClick={this.buttonClick}>=</button>
          </div>
        </div>
      </div>
    );
  }
}

// export default Calculator;
