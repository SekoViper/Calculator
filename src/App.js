import Calculator from './Components/Calculator';
import './style.css';

function App() {
  return (
    <div className="App">
      <Calculator />
    </div>
  );
}

export default App;
