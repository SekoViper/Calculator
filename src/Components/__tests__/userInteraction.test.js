import { render, screen, fireEvent } from '@testing-library/react';
import Renderer from 'react-test-renderer';
import { BrowserRouter, Router } from 'react-router-dom';
import { createMemoryHistory } from '@remix-run/router';
import Home from '../Home';
import Quotes from '../Quotes';
import Calculator from '../Calculator';
import '@testing-library/jest-dom';
import App from '../../App';

// test for home component
describe('Home component', () => {
  test('Home Component is rendered', () => {
    render(<Home />);
    const cont = screen.getByText(/Welcome/);
    expect(cont).toBeInTheDocument();
  });

  test('Matches Snapshots', () => {
    const domTree = Renderer.create(<Home />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});

