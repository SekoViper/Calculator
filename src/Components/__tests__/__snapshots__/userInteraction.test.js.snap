// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component Matches Snapshots 1`] = `
<div
  className="calculator"
>
  <div
    className="display"
  >
    <span
      className="results"
      data-testid="total"
    >
      0
    </span>
    <span
      className="operation"
    />
    <span
      className="next"
    />
  </div>
  <div
    className="buttons"
  >
    <div
      className="digits"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="span-row"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="s"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </div>
    <div
      className="operators"
    >
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Home component Matches Snapshots 1`] = `
Array [
  <h1>
    Welcome
  </h1>,
  <p>
    It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using Content here, content here, making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for lorem ipsum will uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like).
  </p>,
]
`;

exports[`Quotes component Matches Snapshots 1`] = `
Array [
  <h1>
    Where can I get some
  </h1>,
  <p>
    It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using Content here, content here, making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for lorem ipsum will uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like).
  </p>,
]
`;
